sense-android-library/src/nl/sense_os/datastorageengine/CommonSenseProxy.java

```diff
@@ -0,0 +1,597 @@
+
+        String session_id = response.get("session-id");
+
+        if (result != 0 && session_id == null) {
+            // something went horribly wrong
+            Log.w(TAG, "CommonSense login failed: no session id received!");
+            throw new IOException("login failed with no session id");
+        }
+        return session_id;
+    }
+
+
+    /**
+     * Logout the currently logged in user.
+     *
+     * @param sessionID		The sessionID of the user to logout. Cannot be empty.
+     * @throws IOException
+     * @return				Whether or not the logout finished successfully.
+     */
+    public boolean logoutCurrentUser(String sessionID) throws IOException
+    {
+        if(sessionID == null || sessionID.isEmpty())
+            throw new IOException("invalid input of session ID");
+
+        final String url = URL_AUTH + "/" + URL_LOGOUT;
+        Map<String, String> response = request(url, null, sessionID, HTTP_METHOD_POST);
+        // if response code is not 200 (OK), the logout was incorrect
+        String responseCode = response.get(RESPONSE_CODE);
+        int result = checkResponseCode(response.get(RESPONSE_CODE), "logout");
+        if(result != 0)
+            throw new IOException("logout with session id failed ");
+
+        return (result == 0);
+
+    }
+
+    /**
+     * Create a new sensor in the commonsense backend
+     *
+     * Each sensor in commonsense is uniquely identified by a name and device type combination.
+     * If this combination already exists in the commonsense backend this call will fail.
+     *
+     * @param name			Name of the sensor to create. This will be used to identify the sensor. Required.
+     * @param displayName	Extra field to make sensor name more readable if necessary when displaying it. Not required.
+     * @param deviceType	Name of the device type the sensor belongs to. Required.
+     * @param dataType		Type of data that the sensor stores. Required.
+     * @param dataStructure	Structure of the data in the data; can be used to specify the JSON structure in the sensor. Not required.
+     * @param sessionID		The sessionID of the current user. Cannot be empty.
+     * @throws IOException, JSONException
+     *@result				JSONObject with the information of the created sensor. Null if the sensor ID is null or empty.
+     */
+    public JSONObject createSensor(String name, String displayName, String deviceType, String dataType, String dataStructure, String sessionID) throws IOException, JSONException
+    {
+        if(name == null || name.isEmpty() || sessionID == null || sessionID.isEmpty()|| deviceType == null || deviceType.isEmpty()||  dataType == null || dataType.isEmpty())
+            throw new IOException("invalid input of name or sessionID or deviceType or dataType");
+
+        if(dataStructure == null) {
+            dataStructure = "";
+        }
+
+        if(displayName == null) {
+            displayName = "";
```

Comment:
in iOS, it passes an empty string, that why I did it here. Should we use the name instead ?